pkg/driver/nydus/nydus.go

```diff
@@ -433,76 +436,124 @@ func (d *Driver) getChunkDict(ctx context.Context, provider accelcontent.Provide
 	resolver, err := pvd.Resolver(ref)
 	if err != nil {
-		return err
+		return nil, err
 	}
-
 	rc := &containerd.RemoteContext{
-		Resolver: resolver,
+		Resolver:        resolver,
+		PlatformMatcher: d.platformMC,
 	}
 	name, desc, err := rc.Resolver.Resolve(ctx, ref)
 	if err != nil {
 		if errors.Is(err, containerdErrDefs.ErrNotFound) {
 			// Remote cache may do not exist, just return nil
-			return nil
+			return nil, nil
 		}
-		return err
+		return nil, err
 	}
 	fetcher, err := rc.Resolver.Fetcher(ctx, name)
 	if err != nil {
-		return err
+		return nil, err
 	}
 	ir, err := fetcher.Fetch(ctx, desc)
 	if err != nil {
 		if errdefs.NeedsRetryWithHTTP(err) {
 			pvd.UsePlainHTTP()
 			ir, err = fetcher.Fetch(ctx, desc)
 			if err != nil {
-				return errors.Wrap(err, "try to pull remote cache")
+				return nil, errors.Wrap(err, "try to pull remote cache")
 			}
 		} else {
-			return errors.Wrap(err, "pull remote cache")
+			return nil, errors.Wrap(err, "pull remote cache")
 		}
 	}
+	defer ir.Close()
 
-	bytes, err := io.ReadAll(ir)
+	manifestIndexBytes, err := io.ReadAll(ir)
 	if err != nil {
-		return errors.Wrap(err, "read remote cache to bytes")
+		return nil, errors.Wrap(err, "read remote cache bytes to manifest index")
+	}
+	manifestIndex := ocispec.Index{}
+	if err = json.Unmarshal(manifestIndexBytes, &manifestIndex); err != nil {
+		return nil, err
+	}
+	manifestIndexDesc, _, err := nydusutils.MarshalToDesc(manifestIndex, ocispec.MediaTypeImageIndex)
+	if err != nil {
+		return nil, errors.Wrap(err, "marshal remote cache to manifest index  failed")
+	}
+	if err = content.WriteBlob(ctx, pvd.ContentStore(), ref, bytes.NewReader(manifestIndexBytes), *manifestIndexDesc); err != nil {
+		return nil, errors.Wrap(err, "remote cache write blob failed")
 	}
 
-	// TODO: handle manifest list for multiple platform.
-	manifest := ocispec.Manifest{}
-	if err = json.Unmarshal(bytes, &manifest); err != nil {
-		return err
+	for _, manifest := range manifestIndex.Manifests {
+		mDesc := ocispec.Descriptor{
+			MediaType: manifest.MediaType,
+			Digest:    manifest.Digest,
+			Size:      manifest.Size,
+		}
+		mir, err := fetcher.Fetch(ctx, mDesc)
+		if err != nil {
+			return nil, errors.Wrap(err, "fetch remote cache")
+		}
+		manifestBytes, err := io.ReadAll(mir)
+		if err != nil {
+			return nil, errors.Wrap(err, "read remote cache bytes to manifest")
+		}
+		if err = content.WriteBlob(ctx, pvd.ContentStore(), ref, bytes.NewReader(manifestBytes), mDesc); err != nil {
+			return nil, errors.Wrap(err, "remote cache write blob failed")
+		}
+
 	}
 
+	matchDescs, err := utils.GetManifests(ctx, pvd.ContentStore(), *manifestIndexDesc, d.platformMC)
+	if err != nil {
+		return nil, errors.Wrap(err, "get remote cache manifest list")
+	}
+	if sourceDesc.Platform == nil {
+		sourceDesc.Platform = &ocispec.Platform{
+			Architecture: runtime.GOARCH,
```

Comment:
There is a serious problem here. For example if we push an `arm64` image on an `amd` plantform, the cache desc's `Plantform` still will be written `amd` , so we may need another method for those image which lost `Plantform` member .

Reply:
Maybe read `config` to get `arch` and `os` of an image?

Reply:
Indeed, can we find the solution in containerd codes?

Reply:
https://github.com/containerd/containerd/blob/c13f47a3aeab1c01e7660c6c6ab473e27c1d01ac/images/image.go#L261C33-L261C33
How about this?